[Enhancement]: Support Docker desktop on Linux

### Module

Core

### Proposal

Currently, when I try to run Testcontainers on Linux with Docker desktop installed I get this error:

```
2023-01-20T13:06:37.525GMT+02:00 INFO  o.testcontainers.DockerClientFactory  - 0 0 - Connected to docker: 
  Server Version: 20.10.21
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 4178 MB
2023-01-20T13:06:37.595GMT+02:00 INFO  🐳 [testcontainers/ryuk:0.3.3]  - 0 0 - Creating container for image: testcontainers/ryuk:0.3.3
2023-01-20T13:06:37.613GMT+02:00 INFO  o.t.utility.RegistryAuthLocator  - 0 0 - Credential helper/store (docker-credential-desktop) does not have credentials for https://index.docker.io/v1/
2023-01-20T13:06:37.745GMT+02:00 INFO  🐳 [testcontainers/ryuk:0.3.3]  - 0 0 - Container testcontainers/ryuk:0.3.3 is starting: 71cb2840729392cf9a66d23edfae5dcfcc2b5f36d9bf6e30e99052019455191f
2023-01-20T13:06:38.182GMT+02:00 ERROR 🐳 [testcontainers/ryuk:0.3.3]  - 0 0 - Could not start container
com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"No such container: 71cb2840729392cf9a66d23edfae5dcfcc2b5f36d9bf6e30e99052019455191f"}

```

Ok, I hit a dead end.

I tried adding the following to the **.testcontainers.properties** - `docker.host=unix:///home/vcvitaly/.docker/desktop/docker.sock`

With that mount the ryuk container fails to start with the following error:
```
2023/02/02 17:21:20 Pinging Docker...
panic: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

goroutine 1 [running]:
main.main()
	/go/src/github.com/testcontainers/moby-ryuk/main.go:39 +0x4e8
```

However when I do `sudo ln -s /home/vcvitaly/.docker/desktop/docker.sock /var/run/` and clean up the properties, it mounts the symlink and is able to connect to the socket. This is something that experienced Linux users might be able to explain...

The workaround remains:
```
sudo ln -s /home/<YOUR_USERNAME>/.docker/desktop/docker.sock /var/run/
```
This is a brittle solution however since /var/run must be cleaned up on every restart and besides making this symlink requires Docker Desktop to be running so I'm not even sure that you can run this workaround on start-up.

--- a/core/src/main/java/org/testcontainers/dockerclient/RootlessDockerClientProviderStrategy.java
+++ b/core/src/main/java/org/testcontainers/dockerclient/RootlessDockerClientProviderStrategy.java
@@ -34,8 +34,15 @@ public final class RootlessDockerClientProviderStrategy extends DockerClientProv
                 Path homePath = Paths.get(System.getProperty("user.home")).resolve(".docker").resolve("run");
                 return tryFolder(homePath)
                     .orElseGet(() -> {
-                        Path implicitPath = Paths.get("/run/user/" + LibC.INSTANCE.getuid());
-                        return tryFolder(implicitPath).orElse(null);
+                        Path dockerDesktopPath = Paths
+                            .get(System.getProperty("user.home"))
+                            .resolve(".docker")
+                            .resolve("desktop");
+                        return tryFolder(dockerDesktopPath)
+                            .orElseGet(() -> {
+                                Path implicitPath = Paths.get("/run/user/" + LibC.INSTANCE.getuid());
+                                return tryFolder(implicitPath).orElse(null);
+                            });
                     });
             });
     }
@@ -79,7 +86,11 @@ public final class RootlessDockerClientProviderStrategy extends DockerClientProv
 
     @Override
     protected boolean isApplicable() {
-        return SystemUtils.IS_OS_LINUX && getSocketPath() != null && Files.exists(getSocketPath());
+        return (
+            (SystemUtils.IS_OS_LINUX || SystemUtils.IS_OS_MAC) &&
+            getSocketPath() != null &&
+            Files.exists(getSocketPath())
+        );
     }
 
     @Override

